dev/validate_bom__test.py

```diff
@@ -593,6 +632,44 @@ def wait_on_service(self, service_name, port=None, timeout=None):
         ResponseError('It appears that {0} failed'.format(service_name),
                       server='tunnel'))
 
+  def __wait_on_service_using_host(self, service_name, timeout=None):
+    service_config = self.__outbound_service_configs[service_name]
+    host = service_config['host']
+
+    timeout = timeout or self.options.test_service_startup_timeout
+    end_time = time.time() + timeout
+    logging.info('Waiting on "%s..."', service_name)
+
+    threadid = threading.current_thread().ident
+    logging.info('WaitOn polling %s from thread %x', service_name, threadid)
+    try:
+      if 'bearer_auth_token' in service_config:
+        _make_get_request_with_bearer_auth_token('{host}/health'
+                                            .format(host=host),
+                                            service_config['bearer_auth_token'])
+      # Add more authentication cases here.
+      else:
+        urlopen('{host}/health'
+                .format(host=host))
+      logging.info('"%s" is ready on service host %s | %s',
+                   service_name, host, threadid)
+      return
+    except HTTPError as error:
+      logging.error('%s service host got %s.',
+                    service_name, error)
+      raise_and_log_error(
+          ResponseError('{0} service host got {1}'.format(service_name, error),
+                        server=host))
+    except Exception as error:
+      raise_and_log_error(
+          ResponseError('{0} service host got {1}'.format(service_name, error),
+                        server=host))
+    except URLError as error:
+      if time.time() >= end_time:
+        logging.error(
+            'Timing out waiting for %s | %s', service_name, threadid)
```

Comment:
threadid=%x

It seems like you call this one time and not i a loop. So if you have another URL error in here, you are going to eat it whereas in a polling loop you'd retry and perhaps eventually timeout and throw then.